allennlp/training/metrics/auc.py

```diff
@@ -82,7 +87,28 @@ def __call__(
             [self._all_gold_labels, torch.masked_select(gold_labels, mask).long()], dim=0
         )
 
+        if is_distributed():
+            try:
+                # The following logic will only work if the batches are of equal length.
+                world_size = dist.get_world_size()
+                _all_predictions = [
+                    torch.zeros(self._all_predictions.shape) for i in range(world_size)
+                ]
+                _all_gold_labels = [
+                    torch.zeros(self._all_gold_labels.shape, dtype=torch.long)
+                    for i in range(world_size)
+                ]
+                dist.all_gather(_all_predictions, self._all_predictions)
+                dist.all_gather(_all_gold_labels, self._all_gold_labels)
```

Comment:
Potential memory concern with concatenating tensors across processes.

Reply:
I think with one float per instance, we're OK here.